src/utils/ParseRouteUtils.js

```diff
@@ -222,9 +223,36 @@ async function condenseRoute(
         );
     }
 
+    // Ensure that arrivalTime and departureTime have at least one minute difference
+    adjustRouteTimesIfNeeded(route);
     return route;
 }
 
+/**
+ * If the route arrivalTime and departureTime are within the same minute, increase the
+ * arrivalTime by one minute
+ * @param route
+ */
+function adjustRouteTimesIfNeeded(route: Object) {
+    const departureDate = new Date(route.departureTime);
+    const arrivalDate = new Date(route.arrivalTime);
+    if (!isValidDates(departureDate, arrivalDate)) {
+        route.arrivalTime = `${new Date(
+            arrivalDate.getTime() + ONE_MIN_IN_MILLISECONDS,
+        ).toISOString().split('.')[0]}Z`;
+    }
+}
+
+/**
+ * Returns whether departureDate and arrivalDate are within the same minute
+ * @param route
+ * @returns boolean
+ */
+function isValidDates(departureDate: Date, arrivalDate: Date): boolean {
+    return !(departureDate.getTime() + ONE_MIN_IN_MILLISECONDS > arrivalDate.getTime()
```

Comment:
why is the first check necessary?

Reply:
the minutes could be the same but the hour is different

Reply:
ah you are smart, can we make these explicit as in:
```
const isSameMinute = ...
const isSameHour = ...
return isSameMinute && isSameHour;
```
it'll be easier to understand in the future